### ISSUE TITLE
 impact of the gradient_accumulation_every setting on the model's convergence

### ISSUE THREAD

**Issue by jeanRassaire** [mobaidoctor/med-ddpm#26]
Hello,

I'd like to understand the effect of the gradient_accumulation_every parameter.

From reviewing the piece of code below, it appears that not all batches are utilized for the training.

For example, you've set gradient_accumulation_every=2 in your code, which suggests that only two batches are used for each step. I expected all training batches to be used at each step before moving to the next. So, if I'm interpreting this correctly, with 10 training images, a batch_size of 1, and gradient_accumulation_every=2, then for each step, only two images (accounting for gradient_accumulation_every=2 and batch_size=1) are used to update the gradient. My question is: have you tested the impact of the gradient_accumulation_every setting on the model's convergence?

``` Python
  def train(self):
        backwards = partial(loss_backwards, self.fp16)
        start_time = time.time()
        while self.step < self.train_num_steps:
            accumulated_loss = []
            for i in range(self.gradient_accumulate_every):
                if self.with_condition:
                    data = next(self.dl)
                    input_tensors = data['input'].to('cuda:0')
                    target_tensors = data['target'].to('cuda:0')
                    loss = self.model(target_tensors, condition_tensors=input_tensors)
                else:
                    data = next(self.dl).cuda()
                    loss = self.model(data)
                loss = loss.sum()/self.batch_size
                backwards(loss / self.gradient_accumulate_every, self.opt)
                    
```

**Comment by jeanRassaire** [comment]
thanks so much @mobaidoctor
